### PR TITLE
[#1047] improvement(core): Fix the issue of StringIdentifier in metalake property

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Remove the `stringId` from metalake property map. 

### Why are the changes needed?

`gravitino.identifier` is generated internally and should not be visible to users.  

Fix: #1047 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

Existing UTs can cover it.

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/StringIdentifier.java [hunk 1]
```diff
@@ -108,6 +109,27 @@ public static Map<String, String> addToProperties(
         .build();
   }
 
+  /**
+   * Remove StringIdentifier from properties.
+   *
+   * @param properties the properties to remove the string identifier from.
+   * @return the properties with the string identifier removed.
+   */
+  public static Map<String, String> removeFromProperties(Map<String, String> properties) {
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/StringIdentifier.java:118]: I would rename this method to `removeIdFromProperties`.

File: core/src/main/java/com/datastrato/gravitino/StringIdentifier.java [hunk 2]
```diff
@@ -108,6 +109,27 @@ public static Map<String, String> addToProperties(
...
+    }
+
+    if (!properties.containsKey(ID_KEY)) {
+      return properties;
+    }
+
+    Map<String, String> copy = Maps.newHashMap(properties);
+    copy.remove(ID_KEY);
+
+    return ImmutableMap.copyOf(copy);
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/StringIdentifier.java:130]: This unnecessarily copies the map twice, you can optimize this logic to remove one copy.

File: core/src/main/java/com/datastrato/gravitino/meta/MetalakeManager.java [hunk 3]
```diff
@@ -238,4 +242,15 @@ private BaseMetalake.Builder updateEntity(
 
     return builder.withProperties(newProps);
   }
+
+  private BaseMetalake removeIdFromProperties(BaseMetalake metalake) {
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/meta/MetalakeManager.java:246]: This unnecessarily build the metalake again, can you please optimize this logic.

File: core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java [hunk 4]
```diff
@@ -188,6 +188,7 @@ private void collectAndRemoveOldVersionData() throws IOException {
 
       // All keys in this transaction have been deleted, we can remove the commit mark.
       if (keysDeletedCount == keysInTheTransaction.size()) {
+        LOG.info("Physically delete commit mark: {}", Bytes.wrap(kv.getKey()));
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java:191]: This is an unrelated change, please remove this.

File: core/src/main/java/com/datastrato/gravitino/StringIdentifier.java (hunk 2 above)
Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/StringIdentifier.java:130]: I want to return an immutable map here, if it does not matter, the last copy can be omitted.

File: core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java (hunk 4 above)
Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java:191]: OK

File: core/src/main/java/com/datastrato/gravitino/StringIdentifier.java (hunk 2 above)
Comment (jerryshao) [reply core/src/main/java/com/datastrato/gravitino/StringIdentifier.java:130]: No, you can build the immutable map using `builder`, not by copying.

**Comment by yuqi1129** [comment]
@jerryshao 
Pleas take a good again. Thank you.

**Review by jerryshao — APPROVED** [review]
LGTM.
